ons-spark/spark-concepts/partitions.md

```diff
@@ -0,0 +1,941 @@
+````{tabs}
+```{code-tab} py
+rand_df = rand_df.repartition(1)
+rand_df.rdd.getNumPartitions()
+```
+
+```{code-tab} r R
+
+rand_df <- rand_df %>% sparklyr::sdf_repartition(1)
+rand_df %>% sparklyr::sdf_num_partitions()
+
+```
+````
+
+```plaintext
+1
+```
+
+````{tabs}
+```{code-tab} py
+rand_df = rand_df.repartition(10)
+rand_df.rdd.getNumPartitions()
+```
+
+```{code-tab} r R
+
+rand_df <- rand_df %>% sparklyr::sdf_repartition(10)
+rand_df %>% sparklyr::sdf_num_partitions()
+
+```
+````
+
+```plaintext
+10
+```
+Repartition can be used to increase parallelisation.
+
+The other important difference is that `.repartition()` incurs a *full* shuffle of the DataFrame, meaning it rewrites all the data into the new partitions. Shuffling takes time, especially for large amounts of data, so it's best to avoid shuffling more data than needed. To learn more about shuffling have a look at the **Shuffling article**
+
+On the other hand `.coalesce()` involves moving just some of the data. Let's demonstrate by an example.
+
+Say we have a simple DataFrame where the partitions contain the following numbers:
+    
+    Partition 1: 1, 2, 3
+    Partition 2: 4, 5, 6
+    Partition 3: 7, 8, 9
+    Partition 4: 10, 11, 12
+
+We then decide we want two partitions as opposed to four so we apply `.coalesce(2)` to the DataFrame. The partitions would now contain the following:
+
+    Partition 1: 1, 2, 3 + (7, 8, 9)
+    Partition 2: 4, 5, 6 + (10, 11, 12)
+
+So the data on Partitions 1 and 2 have not been shuffled. We have just moved the data on Partitions 3 to Partition 1 and Partition 4 to Partition 2, this is sometimes referred to as a partial shuffle. The gains from this simple example would be negligible, but scale this process up to millions of rows and then it becomes significant.
+
+It's also possible to use a column name in `.repartition()` or even a number and column name, see documentation for more information. 
+
+## Intermediate partitions in wide operations
+
+Now onto a more complex topic of intermediate partitioing.
+
+We saw earlier that for narrow transformations the number of partitions of input and output DataFrames is the same. We also saw that for wide transformations the number of partitions of the output DataFrame is changed to the value of the `spark.sql.shuffle.partitions` property set in the Spark session. 
+
+But what is inside the partitions of the output DataFrame after a wide operation? This depends on the operation, so in this section we will look at three wide transformations: join, groupby and window functions. Understanding how these functions work is particularly useful when we deal with skewed data, or more specifically skew in the join key, group variable or windows. We will demonstrate this with some skewed data and show that knowing your data can help you make informed decisions on scaling jobs vertically or horizontally, or employing an alternative strategy like a **salted join**.
```

Comment:
I'd split groupby into group by, or quote the function directly (if you change it there's a few places you've used this, so worth changing them all for consistency)

Reply:
done